app/forms/state_file/az_senior_dependents_form.rb

```diff
@@ -0,0 +1,32 @@
+module StateFile
+  class AzSeniorDependentsForm < QuestionsForm
+    attr_accessor :dependents_attributes
+    attr_reader :intake
+
+    def initialize(intake = nil, params = nil)
+      super
+      if params.present?
+        @intake.assign_attributes(dependents_attributes: dependents_attributes.to_h)
+      end
+    end
+
+    def dependents
+      @intake.dependents.select(&:ask_senior_questions?)
+    end
+
+    def save
+      @intake.update!({ dependents_attributes: dependents_attributes.to_h })
+    end
+
+    def valid?
```

Comment:
not sure i'm doing this right. seemed to be something magical/not explicit happening in other similar forms i referenced with dependent attributes

Reply:
I think the cleanest implementation right now is on DobForm https://github.com/codeforamerica/vita-min/blob/main/app/forms/state_file/dob_form.rb#L34

The form object's `valid?` should return the combination of the form's own validations plus the validations of any dependents. Since each dependent is just an ActiveRecord `StateFileDependent` the dependent validations are set up on the model, though they can be conditional to ensure they're not required every place we create/edit a dependent.

So in this case `StateFileDependent` should probably have something like a `validates_inclusion_of :passed_away, in: ['yes', 'no'], on: :senior_dob_form` and within `AzSeniorDependentsForm` we should call `dependent.valid?(:senior_dob_form)`

It might be more elegant if there was a `StateFileDependentSeniorQuestionsForm` wrapping each StateFileDependent and the form classes were better able to merge validation errors between the layers, but we haven't built anything like that for our existing use cases.